app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,36 @@
+class SessionsController < ApplicationController
+  #skip_before_action :require_login, only: [:new, :create]
+
+  def new
+    # shows a view with OAuth sign-in link
+  end
+
+  def create #accepts OAuth information from Spotify, finds or creates a User account, and sets user_id in session
+    auth_hash = request.env['omniauth.auth']
+    if auth_hash[:info][:id]
+      @user = User.find_or_create_from_omniauth(auth_hash)
+      if @user
+        session[:user_id] = @user.id
+        redirect_to suggestions_path
+      else
+        redirect_to suggestions_path, notice: "Failed to save the user"
+      end
+
+    else
+      redirect_to suggestions_path, notice: "Failed to authenticate"
+    end
+
+    #user = User.log_in(params[:email], params[:password])
```

Comment:
it's good practice to remove commented-out code if you're not going to use it :)